### DISCUSSION TITLE
Backend configuration

### DISCUSSION THREAD

**Discussion by OngakuKun** [premake/premake-core#1902] [Q&A]
I am currently at the beginn to write my own Engine/Framework thing using Premake to "build" it :yum: I'll support several backends and serching for a way to configure it. 

> **Basically i just want to have a simple way of configuring the Backends without having to mess around using** ``platforms``

I mean I could possibly just do ``Win32-GLFW`` ``Win32-SDL2`` ``Win32-WinAPI`` but yeah, I'll endup with several ``platform``-options 😓 

Sorry for any Typos 😓 My english-grammar isn't the best :yum:

```lua
workspace "SakuraRayEngine"
	startproject "Testbed"

	configurations
	{
		"Debug",
		"Dev",
		"Release",
	}

	platforms
	{
		"Win32",
		"Win64"
	}
	
	flags
	{
		"MultiProcessorCompile",
		"NoPCH"
	}

bindir = "%{wks.location}/.bin/%{cfg.system}-%{cfg.buildcfg}"
intdir = "%{wks.location}/.bin/.int/%{cfg.system}-%{cfg.buildcfg}"

group "Externals"
	include "ext/glfw.lua"

group ""
	include "SaRaEngine"
	include "Testbed"
```
```lua
project "SaRaEngine"
	kind "StaticLib"
	language "C++"
	cppdialect "C++17"
	staticruntime "on"

	targetdir (bindir .. "/%{prj.name}")
	objdir (intdir .. "/%{prj.name}")

	files
	{
		"src/**.hpp",
		"src/**.cpp"
	}

	includedirs
	{
		"src",
		"%{wks.location}/ext/glfw/include"
	}

	links
	{
		"opengl32.lib",
		"GLFW"
	}

	filter "system:windows"
		systemversion "latest"

		defines
		{
			"SARA_PLATFORM_WINDOWS"
		}
	
	filter "platforms:Win32"
		architecture "x32"

	filter "platforms:Win64"
		architecture "x64"

	filter "configurations:Debug*"
		defines "SARA_DEBUG"
		runtime "Debug"
		symbols "on"

	filter "configurations:Dev*"
		defines "SARA_DEV"
		runtime "Release"
		optimize "on"

	filter "configurations:Release*"
		defines "SARA_RELEASE"
		runtime "Release"
		optimize "on"
```

**Comment by nickclark2016** [comment]
So why are you opposed to dealing with various platforms?

**Reply by OngakuKun** [reply]
It's not really because of the various platform. I really need many variations ``Win32-GLFW-OpenGL`` that's just one example and I want to at least support 3 Systmes (Win, Mac, Linux) each with x86, x64 Arm86, arm64, 3 Window Libraries (GLFW, SDL2, Nativ WinAPI) and the Graphic-Libs OpenGL, OpenGL ES, Vulkan, DX 😓

**Reply by nickclark2016** [reply]
Would you want something where your solution is generated with all backends able to be selected from VS (for example), or would you expect only one backend to be outputted when you run Premake.  If it's the former, I'd recommend a configuration for each platform + Debug/Release/Dev variant.

**Reply by OngakuKun** [reply]
I want to have it as modular as possible with the minimum of waittime, so if everything is inside it's ok, but only the one "wanted" will be included. So for ``Win32-GLFW-OpenGL`` it will be targeting Windows x86 with GLFW and OpenGL for windowing and GFX
